### ISSUE TITLE
Bluetooth Connection Issue

### ISSUE THREAD

**Issue by coreylgorman** [mmwise/sphero_ros#18]
Had problem with Bluetooth connectivity/missing Bluetooth module.  Maybe need to add in the documentation that python-bluez needs installed:

sudo apt-get install python-bluez

**Comment by smfreitas** [comment]
Hi,

Did you actually connect? I keep being refused in pairing, I have python-bluez and follow the pairing steps but it keeps failing. Bluetoothctl also fails to pair. How did you pair? (I know I'm asking a question within a question).

Thanks!

**Comment by coreylgorman** [comment]
I was able to connect and see all but the ‘collision.msg’ (still need to dig in on this). I have a new computer on which I’ve only installed Ubuntu 16.04 and ROS kinetic. I started the sphero.launch in the sphero_bringup node and was then able to publish and subscribe to the topics for odom, velocity, etc.

**Comment by coreylgorman** [comment]
The sphero.launch file executed the Bluetooth connection and did not require me to do anything additional.

**Comment by smfreitas** [comment]
What sphero did you use? I have the SPRK+ and that thing throws a fit when I try to pair. I'm not sure if they gave it additional security or something.

Did you have to still connect in the bluetooth screen and setup the pin to "0000"? Thats specifically where I am stuck, as it fails everytime.

**Comment by coreylgorman** [comment]
No, I just put the sphero into pairing mode and it automatically connected once I launched the spher_bringup node. I am using the standard Sphero (not the SPRK+)

**Comment by asoriano1** [comment]
@smfreitas do you get something new with the SPRK+? I'm in the same point, the SPRK+ doesn't pair with the laptop and the launch file just fails to connect with sphero

**Comment by gazialankus** [comment]
Check forks of this repo. Mine, DLu's, etc.

**Comment by supermhel** [comment]
Hi, i also got some issues on my own. I have a Sphero bolt, and i can’t set it into pairing mode( adding of the robot fails). Did the Sphero launch.file help? It didn’t work by my Sphero
